src/System.Device.Gpio/System/Device/Gpio/Drivers/LibgpiodDriver.cs

```diff
@@ -0,0 +1,195 @@
+            gpioGetCommand.StartInfo.Arguments = $"{GpioChip0} {pinNumber}";
+            gpioGetCommand.StartInfo.RedirectStandardOutput = true;
+            gpioGetCommand.Start();
+            if (WaitForProcessToExit(gpioGetCommand, $"{GpioGet} {GpioChip0} {pinNumber}", TimeoutMilliSeconds))
+            {
+                string valueContents = gpioGetCommand.StandardOutput.ReadToEnd();
+                result = ConvertMachineValueToPinValue(valueContents);
+            }
+            return result;
+        }
+
+        private PinValue ConvertMachineValueToPinValue(string value)
+        {
+            return value != null && value.IndexOf('1') != -1 ? PinValue.High : PinValue.Low;
+        }
+
+        protected internal override void RemoveCallbackForPinValueChangedEvent(int pinNumber, PinChangeEventHandler callback)
+        {
+            throw new NotImplementedException();
+        }
+
+        protected internal override void SetPinMode(int pinNumber, PinMode mode)
+        {
+            if (mode == PinMode.InputPullDown || mode == PinMode.InputPullUp)
+            {
+                throw new PlatformNotSupportedException("This driver is generic so it does not support Input Pull Down or Input Pull Up modes.");
+            }
+            MutableTuple<PinMode, Process> pinModeAndProcess;
+            if (_pinNumberToPinModeAndProcess.TryGetValue(pinNumber, out pinModeAndProcess))
+            {
+                pinModeAndProcess.Value1 = mode;
+            }
+            else
+            {
+                _pinNumberToPinModeAndProcess.Add(pinNumber, new MutableTuple<PinMode, Process>(mode, null));
+            }
+        }
+
+        protected internal override WaitForEventResult WaitForEvent(int pinNumber, PinEventTypes eventType, CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();  
+        }
+
+        protected internal override void Write(int pinNumber, PinValue value)
+        {
+                // --mode=signal would tell the process set the value and wait for SIGINT or SIGTERM which needed when setting HIGH value 
+            string command = value == PinValue.High ? $"--mode=signal {GpioChip0} {pinNumber}=1" : $"{GpioChip0} {pinNumber}=0";
+
+            Process gpioSetCommand = new Process();
+            gpioSetCommand.StartInfo.FileName = GpioSet;
+            gpioSetCommand.StartInfo.Arguments = command;
+
+            MutableTuple<PinMode, Process> pinModeAndProcess;
+            if (_pinNumberToPinModeAndProcess.TryGetValue(pinNumber, out pinModeAndProcess))
+            {
+                Process oldProcess = pinModeAndProcess.Value2;
+                if (oldProcess != null)
+                {
+                    oldProcess.Kill();
+                    oldProcess.WaitForExit(1);
+                }
+                gpioSetCommand.Start(); // we need to start the pocess after killing any existing one for the given pin
```

Comment:
If you need it to exit, then you either need to remove the 1 millisecond timeout above, or you need to test the return value of WaitForExit and throw if it returned false.

Probably you want a longer timeout than 1 millisecond which is almost never going to be satisfied.

Reply:
JFYI this is worked fine with the test generating square wave 10000 times. Before making this update when try to set a value to a pin without killing previous one i was getting error something like "the pin is busy", after this update didn't get any issue with testing square wave generation. My point here is the more this wait time set the lower will be implementation, but of course open for suggestion about setting this value